engine/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/AdvisorEmailAggregator.java

```diff
@@ -0,0 +1,85 @@
+package com.redhat.cloud.notifications.processors.email.aggregators;
+
+import com.redhat.cloud.notifications.models.EmailAggregation;
+import io.vertx.core.json.JsonObject;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.HashMap;
+import java.util.Set;
+
+public class AdvisorDailyDigestEmailPayloadAggregator extends AbstractEmailPayloadAggregator {
+
+    // Notification common
+    private static final String EVENT_TYPE_KEY = "event_type";
+    private static final String EVENTS_KEY = "events";
+    private static final String PAYLOAD_KEY = "payload";
+    private static final String INVENTORY_ID = "inventory_id";
+    private static final String PAYLOAD_RULE_ID = "rule_id";
+
+    // Advisor event payloads
+    private static final String NEW_RECOMMENDATION = "new_recommendation";
+    private static final String RESOLVED_RECOMMENDATION = "resolved_recommendation";
+    private static final String DEACTIVATED_RECOMMENDATION = "resolved_recommendation";
+
+    // Advisor events aggregator
+    private static final String ADVISOR_KEY = "advisor";
+    private static final String NEW_RECOMMENDATIONS = "new_recommendations";
+    private static final String RESOLVED_RECOMMENDATIONS = "resolved_recommendations";
+    private static final String DEACTIVATED_RECOMMENDATIONS = "deactivated_recommendations";
+
+    // Advisor event types
+    private static final String NEW_RECOMMENDATION_EVENT = "new-recommendation";
+    private static final String RESOLVED_RECOMMENDATION_EVENT = "resolved-recommendation";
+    private static final String DEACTIVATED_RULE_EVENT = "deactivated-rule";
+    private static final Set<String> EVENT_TYPES = new HashSet<>(Arrays.asList(
+        NEW_RECOMMENDATION_EVENT, RESOLVED_RECOMMENDATION_EVENT,
+        DEACTIVATED_RULE_EVENT
+    ));
+
+    // Advisor final payload helpers
+    private final HashMap<String, HashSet<String>> newRecommendations = new HashMap<String, HashSet<>>();
+    private final HashMap<String, HashSet<String>> resolvedRecommendations = new HashSet<String, HashSet<>>();
+    private final HashSet<String> deactivatedRecommendations = new HashSet<>();
```

Comment:
This part is causing compilation errors. Here's a fix:
```suggestion
    private final Map<String, Set<String>> newRecommendations = new HashMap<>();
    private final Map<String, Set<String>> resolvedRecommendations = new HashMap<>();
    private final Set<String> deactivatedRecommendations = new HashSet<>();
```
⚠️ You will need a few `import`s at the top of the class for this to work.

Reply:
Got it - thanks!